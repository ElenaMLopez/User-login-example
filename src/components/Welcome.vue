<template>
  <div class="home">
    <h1 data-cy="home-title">This is the Home page</h1>
    <p data-cy="home-loged">You are logged.</p>
    <p data-cy="home-logout-info">You can log-out by pressing this button below</p>
    <button class="secondarybtn" @click="_logOut" data-cy="log-out-button">Log Out</button>
  </div>
</template>

<script>
export default {
  name: "Home",
  methods: {
    _logOut() {
      localStorage.clear();
      this.$router.push("/login");
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="scss" scoped>
$primary-color-dark: #0e3669;
.home {
  color: $primary-color-dark;
}
.secondarybtn {
  background: #36a0f7;
  border-radius: 5px;
  border: none;
  color: white;
  font-weight: bold;
  margin-top: 30px;
  margin: 20px 0;
  padding: 10px 10px;
  width: 30%;
}

a {
  color: white;
  font-weight: bold;
  text-decoration: none;
}
</style>
