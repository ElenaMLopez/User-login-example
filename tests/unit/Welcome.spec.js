require("./mock-local-storage");

import { shallowMount, mount } from "@vue/test-utils";
import Welcome from "@/components/Welcome.vue";

describe("Welcome component from Home view", () => {
  it("Should render component", () => {
    const wrapper = shallowMount(Welcome);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should have a logout button", () => {
    const wrapper = shallowMount(Welcome);
    const logoutButton = wrapper.find(".secondarybtn");
    expect(logoutButton.exists()).toBe(true);
  });

  describe("Test click on logout button", () => {
    it("Click on Should call to logout function", () => {
      localStorage.setItem("fake_token", "token");

      const $router = { push: jest.fn() };
      const wrapper = mount(Welcome);
      wrapper.vm.$router = $router;
      wrapper.find(".secondarybtn").trigger("click");

      expect(localStorage.getItem("fake_token")).toBeNull();
    });
  });
});
