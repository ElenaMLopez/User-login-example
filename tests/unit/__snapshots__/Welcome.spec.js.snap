// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component from Home view Should render component 1`] = `
<div class="home">
  <h1 data-cy="home-title">This is the Home page</h1>
  <p data-cy="home-loged">You are logged.</p>
  <p data-cy="home-logout-info">
    You can log-out by pressing this button below
  </p> <button data-cy="log-out-button" class="secondarybtn">
    Log Out
  </button>
</div>
`;
